src/app/shared/api-locator.service.ts

```diff
@@ -33,27 +34,36 @@ export class ApiLocatorService {
   }
 
   get witApiUrl(): string {
-    return this.buildApiUrl('wit');
+    return this.buildApiUrl('wit', (config) => config.witApiUrl);
   }
 
   get forgeApiUrl(): string {
-    let tmp=this.buildApiUrl('forge');
+    let tmp=this.buildApiUrl('forge', (config) => config.forgeApiUrl);
     return tmp
   }
 
   get ssoApiUrl(): string {
-    return this.buildApiUrl('sso');
+    return this.buildApiUrl('sso', (config) => config.ssoApiUrl);
   }
 
   get recommenderApiUrl(): string {
-    return this.buildApiUrl('recommender');
+    return this.buildApiUrl('recommender', (config) => config.recommenderApiUrl);
   }
 
   private loadEnvVar(key: string): void {
     this.envVars.set(key, process.env[this.DEFAULT_API_ENV_VAR_NAMES.get(key)]);
   }
 
-  private buildApiUrl(key: string): string {
+  private buildApiUrl(key: string, fn: (OAuthConfig) => string): string {
```

Comment:
This should just replace the process.env map, rather than suplement it - no need to have two ways to do the same thing.